### PR TITLE
🐛 Fix: cannot get accelerator spec info in perf tuning

### PR DESCRIPTION
**Pull request by trajepl**
## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Format your code by running `pre-commit run --all-files`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: olive/passes/onnx/perf_tuning.py [hunk 1]
```diff
@@ -332,11 +332,11 @@ def _default_config(accelerator_spec: AcceleratorSpec) -> Dict[str, PassConfigPa
         }
 
     def _run_for_config(self, model: ONNXModel, config: Dict[str, Any], output_model_path: str) -> ONNXModel:
-        if "providers_list" not in config:
+        if not config.get("providers_list"):
             # add the provider to the config if user doesn't provide the execution providers
             config["providers_list"] = [self._accelerator_spec.execution_provider]
```
Comment (devang-ml) [inline olive/passes/onnx/perf_tuning.py:337]: User should not have need to provide separate providers_list to ORTPerfTuning pass. We should remove that ability.

The ORTPerfTuning (and other places) should only get EP info from AcceleratorSpec. AcceleratorSpec should be the only place where the user can specify EP(s)

Comment (guotuofeng) [reply olive/passes/onnx/perf_tuning.py:337]: I think the providers_list is used for standalone purpose. for example, the ORTPerfTuning could be used without engine.

**Comment by guotuofeng** [comment]
what's the bug? could its description be more specific?

**Comment by trajepl** [comment]
> what's the bug? could its description be more specific?

`providers_list` and `device` are not required and the default value are None and `cpu`. It means that the check of `providers_list not in config` is always `False`.

```python
if "providers_list" not in config: # <--- the config always have the key of providers_list which is optional in default_config, default value is None
    # add the provider to the config if user doesn't provide the execution providers
    config["providers_list"] = [self._accelerator_spec.execution_provider]

if "device" not in config:
    config["device"] = self._accelerator_spec.accelerator_type
```

You can see that even I give the TRT ep in engine, olive is still use cpu to run perf tuning.
image

**Pull request merged**